snake_case fails semistandard checks

As of semistandard v8 [snake_case is an error](https://github.com/Flet/semistandard/issues/111):

```
> lambda-taggable-createdocument@2.0.2 lint /Users/hassy/workspaces/tc/lambda-taggable-createDocument
> semistandard  --verbose | snazzy

semistandard: Semicolons For All! (https://github.com/Flet/semistandard)

/Users/hassy/workspaces/tc/lambda-taggable-createDocument/test/handler.test.js
  3:5  error  Identifier 'test_hotel_tag' is not in camel case  camelcase
  4:5  error  Identifier 'fields_list' is not in camel case     camelcase

/Users/hassy/workspaces/tc/lambda-taggable-createDocument/test/index.test.js
  5:5  error  Identifier 'test_hotel_tag' is not in camel case  camelcase

/Users/hassy/workspaces/tc/lambda-taggable-createDocument/test/s3_create.test.js
   4:5   error  Identifier 's3_create' is not in camel case  camelcase
   5:5   error  Identifier 'base_url' is not in camel case   camelcase
  28:32  error  Identifier 'base_url' is not in camel case   camelcase
  46:32  error  Identifier 'base_url' is not in camel case   camelcase
```

We should convert the rest of the codebase to camelCase gradually as well.

--- a/index.js
+++ b/index.js
@@ -10,9 +10,17 @@ exports.handler = function (event, context) {
   AwsHelper.init(context, event);
 
   handler.initTagDoc(event, function (err, newTagDoc) {
-    AwsHelper.failOnError(err, event, context);
-    handler.s3_create(newTagDoc, function (err, data) {
-      AwsHelper.failOnError(err, event, context);
+    /* istanbul ignore if */
+    if (err) {
+      return context.fail(err);
+    }
+
+    handler.s3Create(newTagDoc, function (err, data) {
+      /* istanbul ignore if */
+      if (err) {
+        return context.fail(err);
+      }
+
       AwsHelper.log.info({ tag: {eventId: event._id, location: data.Location} }, 'Tag saved to s3');
       return context.succeed(newTagDoc);
     });

--- a/index.js
+++ b/index.js
@@ -10,13 +10,11 @@ exports.handler = function (event, context) {
   AwsHelper.init(context, event);
 
   handler.initTagDoc(event, function (err, newTagDoc) {
-    /* istanbul ignore if */
     if (err) {
       return context.fail(err);
     }
 
     handler.s3Create(newTagDoc, function (err, data) {
-      /* istanbul ignore if */
       if (err) {
         return context.fail(err);
       }

--- a/lib/geonames_lambda.js
+++ b/lib/geonames_lambda.js
@@ -3,7 +3,7 @@ var AWS = require('aws-sdk');
 AWS.config.region = 'eu-west-1';
 var lambda = new AWS.Lambda();
 
-module.exports = function get_geo_tags (record, callback) {
+module.exports = function getGeoTags (record, callback) {
   var params = { // this requires the correct ARN in your Environment Variables
     FunctionName: 'tag-e-geo-v1', // lambda function to invoke
     InvocationType: 'RequestResponse',

--- a/lib/handler.js
+++ b/lib/handler.js
@@ -1,5 +1,5 @@
 'use strict';
-var geonames_lambda = require('./geonames_lambda');
+var geonamesLambda = require('./geonames_lambda');
 
 exports.initTagDoc = function (event, callback) {
   var tag = {
@@ -15,21 +15,21 @@ exports.initTagDoc = function (event, callback) {
   };
   // to get geonames tag/hierarchy we need to trigger the tag-e-geo lambda
   if (tag.location && tag.location.lat && tag.location.lon && tag._id.match(/hotel:mhid/)) {
-    geonames_lambda(tag, function (err, data) {
+    geonamesLambda(tag, function (err, data) {
       if (!err && !data.FunctionError) {
         var res = JSON.parse(data.Payload); // parse the result returned by Geo Lambda
         var g = res[res.length - 1];        // get the last geo tag in the array
         if (tag.tags && tag.tags.length > 0) { //
-          var geo_tag_index = -1; // no geo tag exists
+          var geoTagIndex = -1; // no geo tag exists
           tag.tags.forEach((t, index) => { // a Hotel tag should only have a single Geo Tag
             // over-write the existin geonames tag
             if (t.node.match(/geonames/)) {
-              geo_tag_index = index;
+              geoTagIndex = index;
             }
           });
         }
-        if (geo_tag_index > -1) { // over-write existing geo tag in tags array
-          tag.tags[geo_tag_index] = {
+        if (geoTagIndex > -1) { // over-write existing geo tag in tags array
+          tag.tags[geoTagIndex] = {
             'node': g._id,
             'edge': 'LOCATED_IN',
             'displayName': g.displayName,
@@ -53,4 +53,4 @@ exports.initTagDoc = function (event, callback) {
   }
 };
 
-exports.s3_create = require('./s3_create');
+exports.s3Create = require('./s3_create');

--- a/lib/s3_create.js
+++ b/lib/s3_create.js
@@ -4,7 +4,7 @@ var AWS = require('aws-sdk');
 AWS.config.region = process.env.AWS_REGION;
 var s3bucket = new AWS.S3({params: {Bucket: process.env.AWS_S3_BUCKET}});
 
-module.exports = function s3_create (record, callback) {
+module.exports = function s3Create (record, callback) {
   var type = record._id.split(':')[0];
   var subtype = record._id.split(':')[1].split('.')[0];
   var filepath = type + '/' + subtype + '/' + record._id + '.json';

--- a/package.json
+++ b/package.json
@@ -33,6 +33,7 @@
     "istanbul": "^0.4.3",
     "mocha": "^2.4.5",
     "semistandard": "^8.0.0",
+    "sinon": "1.17.4",
     "snazzy": "^4.0.0"
   },
   "files_to_deploy": [
@@ -41,8 +42,8 @@
     "/lib",
     ".env"
   ],
-  "lambda_memory":"1536",
-  "lambda_timeout":"30",
+  "lambda_memory": "1536",
+  "lambda_timeout": "30",
   "pre-commit": [
     "semistandard",
     "check-coverage"

--- a/test/geonames_lambda.test.js
+++ b/test/geonames_lambda.test.js
@@ -1,6 +1,6 @@
 var assert = require('assert');
 var geo = require('../lib/geonames_lambda');
-var test_hotel_tag = require('./fixtures/test_hotel_tag');
+var testHotelTag = require('./fixtures/test_hotel_tag');
 var AwsHelper = require('aws-lambda-helper');
 
 describe('invoke the Geonames Lambda (tag-e-geo)', function () {
@@ -10,8 +10,9 @@ describe('invoke the Geonames Lambda (tag-e-geo)', function () {
     }, {});
     done();
   });
+
   it('invokes the lambda with sample test_hotel_tag', function (done) {
-    geo(test_hotel_tag, function (err, data) {
+    geo(testHotelTag, function (err, data) {
       if (err || data.Payload.match(/errorMessage/)) console.log(err);
       // console.log(err, data.Payload);
       var res = JSON.parse(data.Payload);
@@ -21,8 +22,8 @@ describe('invoke the Geonames Lambda (tag-e-geo)', function () {
   });
 
   it('invokes the lambda invalid tag (no location.lat)', function (done) {
-    var bad_tag = {_id: 'bad'};
-    geo(bad_tag, function (err, data) {
+    var badTag = {_id: 'bad'};
+    geo(badTag, function (err, data) {
       if (err) console.log(err);
       // console.log(err, data);
       assert.equal(data.FunctionError, 'Handled');

--- a/test/handler.test.js
+++ b/test/handler.test.js
@@ -1,15 +1,15 @@
 var assert = require('assert');
 var handler = require('../lib/handler.js');
-var test_hotel_tag = require('./fixtures/test_hotel_tag.json');
-var fields_list = ['_id', 'location', 'displayName', 'tags',
+var testHotelTag = require('./fixtures/test_hotel_tag.json');
+var fieldsList = ['_id', 'location', 'displayName', 'tags',
   'metadata', 'content', 'markets', 'active', 'description'];
 
 describe('lib/handler.js', function () {
   it('initTagDoc: should create a doc object from the event', function (done) {
-    handler.initTagDoc(test_hotel_tag, function (err, doc) {
+    handler.initTagDoc(testHotelTag, function (err, doc) {
       if (err) console.log(err);
       // console.log(doc);
-      assert.deepEqual(doc, test_hotel_tag);
+      assert.deepEqual(doc, testHotelTag);
       done();
     });
   });
@@ -18,7 +18,7 @@ describe('lib/handler.js', function () {
     var event = { _id: '12345' };
     handler.initTagDoc(event, function (err, doc) {
       if (err) console.log(err);
-      assert.deepEqual(Object.keys(doc), fields_list);
+      assert.deepEqual(Object.keys(doc), fieldsList);
       assert.deepEqual(doc.tags, []);
       assert.deepEqual(doc.metadata, []);
       assert.equal(doc.displayName, event._id);
@@ -31,7 +31,7 @@ describe('lib/handler.js', function () {
     handler.initTagDoc(event, function (err, doc) {
       if (err) console.log(err);
       // console.log(doc);
-      assert.deepEqual(Object.keys(doc), fields_list);
+      assert.deepEqual(Object.keys(doc), fieldsList);
       assert.equal(doc.tags.length, 1);
       assert.equal(doc.tags[0].node, 'geo:geonames.2634643');
       assert.equal(doc.displayName, event._id);
@@ -44,7 +44,7 @@ describe('lib/handler.js', function () {
     handler.initTagDoc(event, function (err, doc) {
       if (err) console.log(err);
       // console.log(doc);
-      assert.deepEqual(Object.keys(doc), fields_list);
+      assert.deepEqual(Object.keys(doc), fieldsList);
       // console.log(doc.tags);
       assert(doc.tags.length > 0);
       assert.equal(doc.tags[1].node, 'geo:geonames.261702');

--- a/test/index.test.js
+++ b/test/index.test.js
@@ -6,6 +6,8 @@ var testHotelTag = require('./fixtures/test_hotel_tag.json');
 var ctxOpts = {
   invokedFunctionArn: 'arn:aws:lambda:eu-west-1:123456789:function:LambdaTest:ci'
 };
+var sinon = require('sinon');
+var handler = require('../lib/handler');
 
 describe('Index handler tests', function () {
   it('Context.fail: called when no id is provided in the event', function (done) {
@@ -24,6 +26,35 @@ describe('Index handler tests', function () {
         done();
       },
       invokedFunctionArn: ctxOpts.invokedFunctionArn
+    };
+    index.handler(testHotelTag, context);
+  });
+
+  it('Lambda should fail if a tag doc fails to be created', function (done) {
+    var mock = sinon.mock(handler);
+    mock.expects('initTagDoc').once().callsArgWith(1, new Error('fake initTagDoc error'));
+    var context = {
+      invokedFunctionArn: ctxOpts.invokedFunctionArn,
+      fail: function (err) {
+        assert(err);
+        mock.verify();
+        return done();
+      }
+    };
+    index.handler(testHotelTag, context);
+  });
+
+  it('Lambda should fail if uploading to S3 fails', function (done) {
+    var mock = sinon.mock(handler);
+    mock.expects('s3Create').once().callsArgWith(1, new Error('fake s3Create error'));
+
+    var context = {
+      invokedFunctionArn: ctxOpts.invokedFunctionArn,
+      fail: function (err) {
+        assert(err);
+        mock.verify();
+        return done();
+      }
     };
     index.handler(testHotelTag, context);
   });

--- a/test/index.test.js
+++ b/test/index.test.js
@@ -2,7 +2,7 @@ var assert = require('assert');
 var index = require('../index');
 var utils = require('aws-lambda-test-utils');
 var mockContextCreator = utils.mockContextCreator;
-var test_hotel_tag = require('./fixtures/test_hotel_tag.json');
+var testHotelTag = require('./fixtures/test_hotel_tag.json');
 var ctxOpts = {
   invokedFunctionArn: 'arn:aws:lambda:eu-west-1:123456789:function:LambdaTest:ci'
 };
@@ -20,11 +20,11 @@ describe('Index handler tests', function () {
   it('Context.succeed: called with the newTagDoc', function (done) {
     var context = {
       succeed: function (result) { // here's your test:
-        assert.deepEqual(result, test_hotel_tag);
+        assert.deepEqual(result, testHotelTag);
         done();
       },
       invokedFunctionArn: ctxOpts.invokedFunctionArn
     };
-    index.handler(test_hotel_tag, context);
+    index.handler(testHotelTag, context);
   });
 });

--- a/test/s3_create.test.js
+++ b/test/s3_create.test.js
@@ -1,8 +1,8 @@
 require('decache')('aws-lambda-helper'); // fresh start
 require('decache')('../lib/s3_create'); // delete cache for test
 var assert = require('assert');
-var s3_create = require('../lib/s3_create');
-var base_url = 'https://numo-taggy.s3-eu-west-1.amazonaws.com/ci/';
+var s3Create = require('../lib/s3_create');
+var baseUrl = 'https://numo-taggy.s3-eu-west-1.amazonaws.com/ci/';
 var AwsHelper = require('aws-lambda-helper');
 
 describe('Save record to S3', function () {
@@ -19,13 +19,13 @@ describe('Save record to S3', function () {
       _id: 'test:testing.123'
     };
 
-    s3_create(obj, function (err, data) {
+    s3Create(obj, function (err, data) {
       if (err) {
         console.log('Error uploading data: ', err);
       } else {
         console.log(data);
       }
-      assert(data.Location === base_url + 'test/testing/test%3Atesting.123.json');
+      assert(data.Location === baseUrl + 'test/testing/test%3Atesting.123.json');
       done();
     });
   });
@@ -36,14 +36,14 @@ describe('Save record to S3', function () {
       _id: 'test:ne.wvHotelPartId.1234'
     };
 
-    s3_create(obj, function (err, data) {
+    s3Create(obj, function (err, data) {
       if (err) {
         console.log('Error uploading data: ', err);
       } else {
         console.log(data);
       }
       // see: https://github.com/numo-labs/taggable-master-hotel-mapping-script/issues/2
-      assert(data.Location === base_url + 'test/ne/test%3Ane.wvHotelPartId.1234.json');
+      assert(data.Location === baseUrl + 'test/ne/test%3Ane.wvHotelPartId.1234.json');
       done();
     });
   });

